JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt

```diff
@@ -54,42 +54,51 @@ fun JetnewsApp(
             val navigationActions = remember(navController) { JetnewsNavigationActions(navController) }
 
             val coroutineScope = rememberCoroutineScope()
-            // This top level scaffold contains the app drawer, which needs to be accessible
+            // This top level composable contains the app drawer, which needs to be accessible
             // from multiple screens. An event to open the drawer is passed down to each
             // screen that needs it.
-            val scaffoldState = rememberScaffoldState()
+            val drawerState = rememberDrawerState(DrawerValue.Closed)
 
             val navBackStackEntry by navController.currentBackStackEntryAsState()
             val currentRoute = navBackStackEntry?.destination?.route ?: JetnewsDestinations.HOME_ROUTE
 
             BoxWithConstraints {
                 val windowSize = getWindowSize(maxWidth)
-                Scaffold(
-                    scaffoldState = scaffoldState,
-                    // If the window size is Compact, show the AppDrawer.
-                    // Otherwise, a NavRail will be shown in individual screens
-                    drawerContent = if (windowSize == WindowSize.Compact) {
-                        {
-                            AppDrawer(
-                                currentRoute = currentRoute,
-                                navigateToHome = navigationActions.navigateToHome,
-                                navigateToInterests = navigationActions.navigateToInterests,
-                                closeDrawer = {
-                                    coroutineScope.launch { scaffoldState.drawerState.close() }
-                                }
-                            )
-                        }
-                    } else {
-                        null
-                    }
-                ) { innerPaddingModifier ->
+
+                val allowDrawerToBeShown = windowSize == WindowSize.Compact
```

Comment:
```suggestion
                val allowDrawerToBeShown = remember(windowSize) {
                    windowSize == WindowSize.Compact
                }
```

I'd probably remember this as this composable can be recomposed a lot of times due to `BoxWithConstraints`

Reply:
I'd hope this simple equality check isn't costly enough that it need to be remembered...
If we feel like we need to remember that, wouldn't we basically be trying to remember all logic that's in the main composable scope?

Reply:
Sure, that's ok